rails_application/app/controllers/products_controller.rb

```diff
@@ -91,8 +91,8 @@ def set_future_product_price(product_id, price, valid_since)
     command_bus.(set_product_future_price_cmd(product_id, price, valid_since))
   end
 
-  def set_product_vat_rate(product_id, vat_rate_code)
-    vat_rate = Taxes::Configuration.available_vat_rates.find{|rate| rate.code == vat_rate_code}
+  def set_product_vat_rate(product_id, vat_rate)
+    vat_rate = VatRates::AvailableVatRate.find_by!(code: vat_rate).to_value
```

Comment:
I don't think that the controller should validate if the vat rate exists

Reply:
Especially as it happens in the domain model (which is also not correct).

Would you move it to the `SetVatRateHandler`?

Reply:
Done 👍